src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker.cs

```diff
@@ -233,9 +233,12 @@ internal void RemoveProject(AbstractProject project)
 
             UpdateProjectBinPath(project, project.TryGetBinOutputPath(), null);
 
-            if (project.PushingChangesToWorkspaceHosts)
```

Comment:
The bug is that once we had pushed a project to a host, if we removed it, we would never clear it from `WorkspaceHostState._pushedProjects`. Completely closing a solution was fine since we'd clear out everything, but that would leak the reference.

It's unclear to me how "big" this leak actually is; it might be only leaking the immediate AbstractProject and a few other cheap types, but better safe than sorry.